nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,748 @@
+
+```elixir
+v = Nx.tensor([1, 2, 3])
+v1 = ~V(1 2 3)
+v2 = ~V[1 2 3]
+
+{
+  v == v1,
+  v == v2,
+  v
+}
+```
+
+* you can build a matrix or 2D tensor with the sigil `~M`. For example a matrix with [shape](<(https://hexdocs.pm/nx/Nx.html#shape/1)>) (2,3).
+
+```elixir
+m = ~M( 1 2 3
+  4 5 6)
+```
+
+Equivalently, you can write the structure by using nested lists. You can also start with a "flat vector" and _reshape_ it. For example:
+
+```elixir
+m1 = Nx.tensor([[1, 2, 3], [4, 5, 6]])
+
+m2 = ~V(1 2 3 4 5 6) |> Nx.reshape({2, 3})
+
+{
+  m == m1,
+  m == m2,
+  m
+}
+```
+
+* for higher dimensions, you can pass an arbitrarily nested list of numbers, or equivalently, you can reshape a "flat vector" into the dimensions that match your nested list. For example:
+
+```elixir
+t1 = Nx.tensor([[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]])
+
+t2 = ~V(1 2 3 4 5 6 7 8 9 10 11 12) |> Nx.reshape({2, 2, 3})
+
+{
+  Nx.shape(t1),
+  t1 == t2,
+  t1
+}
+```
+
+#### A note on how to access to values of a tensor
+
+If `t` is the tensor, you can [access](https://hexdocs.pm/nx/Nx.html#module-access-syntax-slicing) to its elements by appending the indexes at the end within brackets "[i]" and you get a tensor response. Recall that the indexes start from $0$. For example, with the matrix `t` below, then `t[0][1] = Nx.Tensor(2)}`. You can iterate on the indexes, and access from the end with negative indexes where the index -1 represents the last one.
+
+```elixir
+t = ~M(1 2 3
+  4 5 6)
+
+first_row = t[0]
+first_column = for i <- 0..1, do: t[i][0]
+corners = {t[0][0], t[0][2], t[1][2], t[1][0]}
+reverse_index = t[1][2] == t[-1][-1]
+
+{
+  corners,
+  first_row,
+  first_column,
+  reverse_index
+}
+```
+
+## Aggregations operations on tensors
+
+You can apply [aggregation functions](https://hexdocs.pm/nx/Nx.html#functions-aggregates) on any tensor. The function can be applied to the tensor as a whole or to a subsection of the tensor taken in an axis-wise fashion.
+
+As a first example, let's take a 2D tensor (aka matrix) of shape (2,3). Notice that we can _name_ the axes. The elements of the tensor `t` are the `t[i][j]` when $i$ is the row, and $j$ the column.
+
+```elixir
+m = Nx.tensor([[1, 2, 3], [4, 5, 6]], names: [:x, :y])
+```
+
+You can get the maximum number in a tensor with [Nx.reduce_max(matrix)](https://hexdocs.pm/nx/Nx.html#reduce_max/2), returning a scalar (0D tensor). With a reduction, we loose a dimension per axis reduced, and since we did globally, we loose all the dimensions. It should return `Nx.Tensor(6)` here.
+
+We can get the maximum number for each _row_ with `Nx.reduce_max(matrix, axes: [:y])`, returning a vector (1D tensor) of length 2. Why `:y`? Because for each row, we reduce along the `:y` axis.
+
+We can get the maximum number for each _column_ with `Nx.reduce_max(matrix, axes: [:x])`, returning a vector (1D tensor) of length 3. For each column, we reduce along the axis `:x`.
```

Comment:
```suggestion
We can get the maximum number for each _row_ with `Nx.reduce_max(matrix, axes: [:y])`, returning a 1D tensor of length 2. Why `:y`? Because for each row, we reduce along the `:y` axis.

We can get the maximum number for each _column_ with `Nx.reduce_max(matrix, axes: [:x])`, returning a 1D tensor of length 3. For each column, we reduce along the axis `:x`.
```